app.go

```diff
@@ -151,6 +151,18 @@ func Logger(p Printer) Option {
 	})
 }
 
+// NoopLogger is a no-op logger that implements the Printer interface.
+// When used with the Logger option, all Fx logs will be dropped.
+//
+// An application that wants to disable Fx logs should be created like so:
+//
+// app := fx.New(fx.Logger(fx.NoopLogger{}))
```

Comment:
Would a better experience be:

```
fx.New(fx.NopLogger)
```

In other words, we can make a global option called `NopLogger` instead of having the user instantiate an empty struct.